File: ios/Sources/EmarsysPlugin/EmarsysCore.swift
import Foundation
import EmarsysSDK

@objc public class EmarsysCore: NSObject {
    @objc public func setContact(contactFieldId: Int, contactFieldValue: String, completion: @escaping (Error?) -> Void) {
        Emarsys.setContact(contactFieldId: contactFieldId as NSNumber, contactFieldValue: contactFieldValue) { error in
            completion(error)
        }
    }

    @objc public func trackCustomEvent(eventName: String, eventAttributes: NSDictionary, completion: @escaping (Error?) -> Void) {
        Emarsys.trackCustomEvent(eventName: eventName, eventAttributes: eventAttributes as? [String: String]) { error in
            completion(error)
        }
    }
}

File: android/src/main/java/com/emarsys/plugin/emarsys/EmarsysCore.java
package com.emarsys.plugin.emarsys;

import com.emarsys.Emarsys;

public class EmarsysCore {

    public void setContact(int contactFieldId, String contactFieldValue, EmarsysCompletionListener listener) {
        Emarsys.setContact(contactFieldId, contactFieldValue, error -> {
            listener.onComplete(error);
        });
    }
}
